### PR TITLE
Use clang++ in OSS CI

### PR DESCRIPTION
**Pull request by zasdfgbnm**
We have plenty of pipelines testing gcc, but none of our pipeline is testing clang. Clang is a supported compiler, we should test on it. Also, clang is faster than gcc, so this would also reduce our CI time.

Besides, we also have the following changes:
- `Lint / standalone-build` is renamed as `Build / clang-build`.
- Moved `clang-build` and `dynamic-type-meson` into `build.yml`.

### CODE REVIEW

File: csrc/ir/nodes.cpp
Comment (zasdfgbnm) [inline csrc/ir/nodes.cpp:1]: clang-tidy starts yelling at this file for this PR. This is strange, but fine, let's just fix this file.

**Comment by zasdfgbnm** [comment]
@naoyam I need admin power to change branch protection rule, otherwise this PR can not be merged.

**Comment by zasdfgbnm** [comment]
CI time reduced from `148m` to `102m`.

**Comment by naoyam** [comment]
@xwang233 Does this look good to you? I'll let you stamp.

**Review by xwang233 — COMMENTED** [review]
Thanks for working on the OSS CI setup! Shall we keep both clang and gcc build? Wdyt @naoyam ?

**Comment by naoyam** [comment]
> Thanks for working on the OSS CI setup! Shall we keep both clang and gcc build? Wdyt @naoyam ?

I think it'd be too much, but we have seen different results with the two compilers. Is it possible to run both only weekly?

**Comment by xwang233** [comment]
Sure, we can setup extra clang and gcc to only run over the weekend, weekly.

**Review by xwang233 — APPROVED** [review]

**Comment by xwang233** [comment]
I've paused the branch protection rule for `standalone-build`. Let's merge this PR and we'll enable that for the new build action job.

**Pull request merged**